frontend/src/APIClients/AuthAPIClient.ts

```diff
@@ -142,49 +143,55 @@ const refresh = async (refreshFunction: RefreshFunction): Promise<boolean> => {
   }
 };
 
-type GetUserWithVerificationStatusFunction = (
+type GetEmailVerifiedByEmailFunction = (
   variables?:
     | Partial<{
         email: string;
       }>
     | undefined,
 ) => Promise<
   ApolloQueryResult<{
-    userWithVerificationStatusByEmail: Omit<AuthUser, "accessToken">;
+    emailVerifiedByEmail: boolean;
   }>
 >;
 
-const getAuthUser = async (
-  authUser: AuthUser,
-  getUserWithVerificationStatusFunction: GetUserWithVerificationStatusFunction,
-): Promise<AuthUser> => {
-  if (!authUser) {
-    return null;
-  }
-
-  const { data, error } = await getUserWithVerificationStatusFunction({
-    email: authUser.email,
+const getEmailVerified = async (
+  authUserEmail: string,
+  getEmailVerifiedByEmailFunction: GetEmailVerifiedByEmailFunction,
+): Promise<boolean | null> => {
+  const { data, error } = await getEmailVerifiedByEmailFunction({
+    email: authUserEmail,
   });
-  if (error || data.userWithVerificationStatusByEmail === undefined) {
+  if (error || data.emailVerifiedByEmail === undefined) {
     // TODO: add proper frontend logging
     // eslint-disable-next-line no-console
     console.log(error);
     // eslint-disable-next-line no-console
     console.log(data);
-    return authUser;
+    return null;
   }
-  const newAuthUser = {
-    accessToken: authUser.accessToken,
-    ...data.userWithVerificationStatusByEmail,
-  } as AuthUser;
-  localStorage.setItem(AUTHENTICATED_USER_KEY, JSON.stringify(newAuthUser));
-  return newAuthUser;
+  return data.emailVerifiedByEmail;
+};
+
+const updateAuthUserEmailVerified = (
+  emailVerified: boolean,
+  setAuthUser: Dispatch<SetStateAction<AuthUser>>,
+): void => {
+  setAuthUser((prevAuthUser: AuthUser) => {
+    const newAuthUser = {
+      ...prevAuthUser,
+      emailVerified,
+    } as AuthUser;
+    localStorage.setItem(AUTHENTICATED_USER_KEY, JSON.stringify(newAuthUser));
+    return newAuthUser;
+  });
```

Comment:
I don't think it makes sense to have this function located in this file. Why not include this logic directly at the call site? Or if you feel it's necessary to extract it into a function, could you put the function directly in the file where its used? The APIClient files are really just for functionality which makes calls to the backend.